tests/test_objective_funs.py

```diff
@@ -2418,6 +2418,35 @@ def test_objective_no_nangrad_coils(self, objective):
         g = obj.grad(obj.x())
         assert not np.any(np.isnan(g)), str(objective)
 
+    @pytest.mark.unit
+    @pytest.mark.parametrize("helicity", [(1, 0), (1, 1), (0, 1)])
+    def test_objective_no_nangrad_omnigenity(self, helicity):
+        """Omnigenity."""
+        surf = FourierRZToroidalSurface.from_qp_model(
+            major_radius=1,
+            aspect_ratio=20,
+            elongation=6,
+            mirror_ratio=0.2,
+            torsion=0.1,
+            NFP=1,
+            sym=True,
+        )
+        eq = Equilibrium(Psi=6e-3, M=4, N=4, surface=surf)
+        field = OmnigenousField(
+            L_B=0,
+            M_B=2,
+            L_x=0,
+            M_x=0,
+            N_x=0,
```

Comment:
I'm not certain it will make a difference, but we should increase these resolutions to be non-zero. If the coordinate mapping is constant then there aren't actually any free parameters involved in the part of the code that was causing the NaNs. 

Also you are using a QP equilibrium and testing with OH/OT but I think that should be fine for only testing nan grad